### PR TITLE
DOMA-1615 more mobile api upgrades

### PR DESCRIPTION
**Pull request by toplenboren**
# Problem

1. We decided to make validations required for registerServiceConsumer case
2. We decided to send organizationId instead of tin to the registerServiceConsumer mutation

# Solution

- You can't create ServiceConsumer if you dont have any billingAccount and any Meters 
- Now you need to send organizationId instead of tin to the registerServiceConsumer

### CODE REVIEW

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/domains/resident/schema/RegisterServiceConsumerService.js [hunk 1]
```diff
@@ -58,62 +56,37 @@ const RegisterServiceConsumerService = new GQLCustomSchema('RegisterServiceConsu
...
+                if (billingIntegrationContext) {
 
-                // User can request explicit validation of some fields. This is required for iOS and Android mobile clients
-                if (Array.isArray(validations)) {
-                    validations.forEach(field => {
-                        if (!get(optionalAttrs, [field], false)) {
-                            throw new Error(`${NOT_FOUND_ERROR}${field}] ${field} is not found, but required in custom validations`)
-                        }
-                    })
+                    const [acquiringIntegrationContext] = await AcquiringIntegrationContext.getAll(context, { organization: { id: organization.id } })
```
Comment (SavelevMatthew) [inline apps/condo/domains/resident/schema/RegisterServiceConsumerService.js:77]: deletedAt?

File: apps/condo/domains/resident/schema/RegisterServiceConsumerService.js [hunk 2]
```diff
@@ -58,62 +56,37 @@ const RegisterServiceConsumerService = new GQLCustomSchema('RegisterServiceConsu
...
                     sender,
                     resident: { connect: { id: residentId } },
                     accountNumber: accountNumber,
                     organization: { connect: { id: organization.id } },
                 }
 
-                const optionalAttrs = {
-                    ...paymentFeatureAttrs,
-                }
+                const [ billingIntegrationContext ] = await BillingIntegrationOrganizationContext.getAll(context, { organization: { id: organization.id } })
```
Comment (SavelevMatthew) [inline apps/condo/domains/resident/schema/RegisterServiceConsumerService.js:74]: deletedAt?

File: apps/condo/domains/resident/schema/RegisterServiceConsumerService.js [hunk 3]
```diff
@@ -58,62 +56,37 @@ const RegisterServiceConsumerService = new GQLCustomSchema('RegisterServiceConsu
...
+                    attrs.billingAccount = billingAccount ? { connect: { id: billingAccount.id } } : null
+                    attrs.billingIntegrationContext = billingAccount ? { connect: { id: billingIntegrationContext.id } } : null
+                    attrs.acquiringIntegrationContext = billingAccount && acquiringIntegrationContext ? { connect: { id: acquiringIntegrationContext.id } } : null
                 }
 
-                const attrs = {
-                    ...requiredAttrs,
-                    ...optionalAttrs,
+                if (!attrs.billingAccount) {
+                    const meters = await Meter.getAll(context, { accountNumber: accountNumber, unitName: unitName, organization: { id: organizationId } })
```
Comment (SavelevMatthew) [inline apps/condo/domains/resident/schema/RegisterServiceConsumerService.js:86]: deletedAt?

File: apps/condo/migrations/20211125182618-0084_auto_20211125_1326.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
...aWdyYXRpb24obWlncmF0aW9ucy5NaWdyYXRpb24pOgoKICAgIGRlcGVuZGVuY2llcyA9IFsKICAgICAgICAoJ19kamFuZ29fc2NoZW1hJywgJzAwODNfYXV0b18yMDIxMTEyM18wODA0JyksCiAgICBdCgogICAgb3BlcmF0aW9ucyA9IFsKICAgICAgICBtaWdyYXRpb25zLkFkZEZpZWxkKAogICAgICAgICAgICBtb2RlbF9uYW1lPSdzZXJ2aWNlY29uc3VtZXInLAogICAgICAgICAgICBuYW1lPSdwYXltZW50Q2F0ZWdvcnknLAogICAgICAgICAgICBmaWVsZD1tb2RlbHMuVGV4dEZpZWxkKGJsYW5rPVRydWUsIG51bGw9VHJ1ZSksCiAgICAgICAgKSwKICAgICAgICBtaWdyYXRpb25zLkFkZEZpZWxkKAogICAgICAgICAgICBtb2RlbF9uYW1lPSdzZXJ2aWNlY29uc3VtZXJoaXN0b3J5cmVjb3JkJywKICAgICAgICAgICAgbmFtZT0ncGF5bWVudENhdGVnb3J5JywKICAgICAgICAgICAgZmllbGQ9bW9kZWxzLlRleHRGaWVsZChibGFuaz1UcnVlLCBudWxsPVRydWUpLAogICAgICAgICksCiAgICAgICAgbWlncmF0aW9ucy5BZGRGaWVsZCgKICAgICAgICAgICAgbW9kZWxfbmFtZT0nc2VydmljZWNvbnN1bWVyaGlzdG9yeXJlY29yZCcsCiAgICAgICAgICAgIG5hbWU9J3Jlc2lkZW50QmlsbGluZ0FjY291bnQnLAogICAgICAgICAgICBmaWVsZD1tb2RlbHMuSlNPTkZpZWxkKGJsYW5rPVRydWUsIG51bGw9VHJ1ZSksCiAgICAgICAgKSwKICAgICAgICBtaWdyYXRpb25zLkFkZEZpZWxkKAogICAgICAgICAgICBtb2RlbF9uYW1lPSdzZXJ2aWNlY29uc3VtZXJoaXN0b3J5cmVjb3JkJywKICAgICAgICAgICAgbmFtZT0ncmVzaWRlbnRPcmdhbml6YXRpb24nLAogICAgICAgICAgICBmaWVsZD1tb2RlbHMuSlNPTkZpZWxkKGJsYW5rPVRydWUsIG51bGw9VHJ1ZSksCiAgICAgICAgKSwKICAgIF0K
```
Comment (SavelevMatthew) [inline apps/condo/migrations/20211125182618-0084_auto_20211125_1326.js:2]: Don't forget to keep your migrations up-to-date

**Review by SavelevMatthew — APPROVED** [review]

**Pull request merged**